modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/component/FileStatusIconsView.kt

```diff
@@ -101,24 +104,39 @@ class FileStatusIconsView @JvmOverloads constructor(
 
         val isInUpNextObservable = upNextChangesObservable.containsUuid(episode.uuid)
 
+        data class CombinedData(
+            val streamingProgress: EpisodeStreamProgress,
+            val playbackState: PlaybackState,
+            val isInUpNext: Boolean,
+            val hasBookmark: Boolean,
+        )
+
         val playbackStateForThisEpisode = playbackStateUpdates
             .startWith(PlaybackState(episodeUuid = episode.uuid)) // Pre load with a blank state so it doesn't wait for the first update
             .filter { it.episodeUuid == episode.uuid } // We only care about playback for this row
 
         disposables.clear()
-        Observables.combineLatest(combinedProgress, playbackStateForThisEpisode, isInUpNextObservable)
+        Observables.combineLatest(
+            combinedProgress,
+            playbackStateForThisEpisode,
+            isInUpNextObservable,
+            bookmarkVisibilityObservable,
+        ) { streamProgress, playbackState, isInUpNext, hasBookmark ->
+            CombinedData(streamProgress, playbackState, isInUpNext, hasBookmark)
+        }
             .distinctUntilChanged()
             .toFlowable(BackpressureStrategy.LATEST)
             .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread())
             .doOnTerminate { UploadProgressManager.stopObservingUpload(episode.uuid, uploadConsumer) }
-            .doOnNext { (streamingProgress, playbackState, isInUpNext) ->
-                episode.playing = playbackState.isPlaying && playbackState.episodeUuid == episode.uuid
-                imgUpNext.visibility = if (isInUpNext) View.VISIBLE else View.GONE
+            .doOnNext { combinedData ->
+                episode.playing = combinedData.playbackState.isPlaying && combinedData.playbackState.episodeUuid == episode.uuid
+                episode.hasBookmark = combinedData.playbackState.episodeUuid == episode.uuid
```

Comment:
As you mentioned, this is not required. When you'll receive the updated bookmarks list, it'll have an updated `hasBookmark` status.